### DISCUSSION TITLE
Release Scopes

### DISCUSSION THREAD

**Discussion by rohanjsh** [rohanjsh/typeset#40] [General]
For v2.1

- [x] #39 
- [x] #38 
- [x] Typeset Context Menus
- [x] Support all the properties for TypesetExtension
- [x] Test coverage to above 95%
- [ ] Support for colors using hex code

**Comment by rohanjsh** [comment]
As I go ahead with TypesetEditingController, first step was to create context menus which can be as simple as plug-n-play.
Below is the initial demo of the same. This is a top-level function defined for the package

**Code**
```dart
TextField(
   contextMenuBuilder: (context, editableTextState) {
   return AdaptiveTextSelectionToolbar.buttonItems(
          anchors: editableTextState.contextMenuAnchors,
          buttonItems: getTypesetContextMenus(
         editableTextState: editableTextState,
        ),
     );
  },
),
```

**Demo**

https://github.com/rohanjsh/typeset/assets/35066779/73310d17-1b80-4ef8-974a-9d918fae1d82


Thoughts @FirdousNath ?

**Reply by rohanjsh** [reply]
Target for TypesetEditingController - Screenshot from whatsapp, where characters are preserved (but greyed out) and text is also styled
![8b824c1b-3e0e-44ea-abdd-5604e2e072c6](https://github.com/rohanjsh/typeset/assets/35066779/a13d96c5-cefa-4cbc-873c-61bea5cb94f5)

**Reply by FirdousNath** [reply]
looks good but only issue is with nested styling and what about font size? Also didn’t we need to show realtime view of styling?

**Reply by rohanjsh** [reply]
We are not going to support nested styling as of now. Have added checks to not show context menu if text is already wrapped with literal chars.
Yes, I am working on realtime view of styling. This is just the initial stage.

**Reply by rohanjsh** [reply]
I dont think font size should be part of context menu.

**Comment by rohanjsh** [comment]
For v2.2

- [ ] TypesetEditingController - similar to whatsapp https://github.com/rohanjsh/typeset/discussions/40#discussioncomment-8024111
